File: src/p1005/golang/solution.go
package main

import (
	"math"
	"sort"
)

func largestSumAfterKNegations(nums []int, k int) int {
	sort.Ints(nums)
	total, value := 0, math.MaxInt32
	for _, num := range nums {
		if num < 0 && k > 0 {
			num, k = -num, k - 1
		}
		total += num
		if value > num {
			value = num
		}
	}
	return total - (k % 2) * 2 * value
}

File: src/p1005/cpp/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;


class Solution {
public:
    int largestSumAfterKNegations(vector<int> &nums, int k) {
        sort(nums.begin(), nums.end());
        auto total = 0, value = INT_MAX;
        for (auto num : nums) {
            if (num < 0 && k > 0) {
                num = -num; k--;
            }
            total += num;
            value = min(value, num);
        }
        return total - (k % 2) * 2 * value;
    }
};
